packages/app/hooks/use-claim-nft.ts

```diff
@@ -106,59 +109,17 @@ const reducer = (state: State, action: Action): State => {
   }
 };
 
-export const useClaimNFT = (edition?: IEdition) => {
+export const useClaimNFT = (edition: IEdition) => {
   const { rudder } = useRudder();
   const router = useRouter();
   const { data: userProfile } = useMyInfo();
-
+  const bottom = usePlatformBottomHeight();
   const signTypedData = useSignTypedData();
-  const [state, dispatch] = useReducer(reducer, initialState);
-  const mutate = useMatchMutate();
+  const { state, dispatch, pollTransaction } = useContext(ClaimContext);
   const Alert = useAlert();
-  const { mutate: mutateEdition } = useCreatorCollectionDetail(
-    edition?.contract_address
-  );
   const { connect } = useWallet();
-
   let { userAddress } = useCurrentUserAddress();
-
-  const pollTransaction = async (transactionId: any) => {
-    let intervalMs = 2000;
-    for (let attempts = 0; attempts < 100; attempts++) {
-      Logger.log(`Checking tx... (${attempts + 1} / 20)`);
-      const response = await axios({
-        url: `/v1/creator-airdrops/poll-mint?relayed_transaction_id=${transactionId}`,
-        method: "GET",
-      });
-      Logger.log(response);
-
-      dispatch({
-        type: "transactionHash",
-        transactionHash: response.transaction_hash,
-      });
-
-      if (response.is_complete) {
-        rudder?.track("NFT Claimed");
-        dispatch({ type: "success", mint: response.mint });
-        mutate((key) => key.includes(PROFILE_NFTS_QUERY_KEY));
-        mutateEdition((d) => {
-          if (d) {
-            return {
-              ...d,
-              is_already_claimed: true,
-              total_claimed_count: d?.total_claimed_count + 1,
-            };
-          }
-        });
-
-        return;
-      }
-
-      await delay(intervalMs);
-    }
-
-    dispatch({ type: "error", error: "polling timed out" });
-  };
+  const snackbar = useSnackbar();
```

Comment:
btw I think should get rid of the hooks code, it always warns in`useEffect` hooks depends.
I will use `createRef` method to call it but improve it needs to the next PR.